pandas/core/common.py

```diff
@@ -84,9 +83,13 @@ def maybe_box(indexer, values, obj, key):
 def maybe_box_datetimelike(value):
     # turn a datetime like into a Timestamp/timedelta as needed
 
-    if isinstance(value, (np.datetime64, datetime)):
+    if isinstance(value, (np.datetime64, datetime)) and (
+            not isinstance(value, tslibs.Timestamp)) and (
+            notnull(value)):
```

Comment:
This shouldn't be isna/notnull, but should be specifically checking for NaT

Reply:
The existing behavior keeps `np.nan` as `np.nan`:
```python
In [1]: import numpy as np; import pandas as pd; pd.__version__
Out[1]: '0.26.0.dev0+1469.ge817ffff3'

In [2]: pd.core.common.maybe_box_datetimelike(np.nan)
Out[2]: nan
```
But fine just checking for `NaT` since it's more in the spirit of what this function is supposed to do.  Either way `np.nan` wouldn't get coerced to `NaT` unless additional modifications are made.

Reply:
i think `np.datetime64("NaT")` would be treated differently